src/main/webapp/app/exercises/text/assess/text-assessment.service.ts

```diff
@@ -110,24 +106,22 @@ export class TextAssessmentService {
         } else {
             params = params.set('correction-round', correctionRound.toString());
         }
-        return this.http
-            .get<StudentParticipation>(`${this.resourceUrl}/submission/${submissionId}`, { observe: 'response', params })
-            .pipe(
-                // Wire up Result and Submission
-                tap((response) => {
-                    const participation = response.body!;
-                    const submission = participation.submissions![0];
-                    let result;
-                    if (resultId) {
-                        result = getSubmissionResultById(submission, resultId);
-                    } else {
-                        result = getSubmissionResultByCorrectionRound(submission, correctionRound)!;
-                    }
-                    TextAssessmentService.reconnectResultsParticipation(participation, submission, result!);
-                    (submission as TextSubmission).atheneTextAssessmentTrackingToken = response.headers.get('x-athene-tracking-authorization') || undefined;
-                }),
-                map((response) => response.body!),
-            );
+        return this.http.get<StudentParticipation>(`${this.resourceUrl}/text-submissions/${submissionId}/for-assessment`, { observe: 'response', params }).pipe(
+            // Wire up Result and Submission
+            tap((response) => {
```

Comment:
While we're here: Would be good if we can specify the type. (Should be `HttpResponse<StudentParticipation>`; same for the `map` below.) Btw, do we even handle errors here?

Since this is only moved code, you don't have to address this comment in this PR though if you don't want to (imo).